### PR TITLE
Remove code related to old HTTP solver API

### PR DESCRIPTION
**Pull request by MartinquaXD**
# Description
Since we removed the last quoter that used the old API we can finally clean up some more stuff.

# Changes
- removed legacy http price estimator from price estimator factory
- deleted http solver
- deleted most of http solver model (and slightly moved around remaining bits)
- deleted legacy adapter and tests
- removed balancerv2 and univ3 subgraph urls from api and autopilot args
- slightly simplified how we construct price estimators in the factory but this can be cleaned up a lot more; just wanted to keep this PR mostly about deleting stuff instead of changing stuff to make it easier to review

## How to test
compiler and e2e tests

### CODE REVIEW

**Review by nlordell — APPROVED** [review]

**Review by Mateo-mro — APPROVED** [review]

File: crates/autopilot/src/run.rs [hunk 1]
```diff
@@ -240,15 +229,12 @@ pub async fn run(args: Arguments) {
             UniV2BaselineSourceParameters::from_baseline_source(*source, &chain_id.to_string())
         })
         .chain(args.shared.custom_univ2_baseline_sources.iter().copied());
-    let (pair_providers, pool_fetchers): (Vec<_>, Vec<_>) = futures::stream::iter(univ2_sources)
+    let pair_providers: Vec<_> = futures::stream::iter(univ2_sources)
```
Comment (fleupold) [inline crates/autopilot/src/run.rs:232]: This is still required for bad token detection? Do we have a plan somewhere on how to remove that beast as well?

**Review by fleupold — APPROVED** [review]

**Comment by fleupold** [comment]
image

Who is this guy?

File: crates/autopilot/src/run.rs (hunk 1 above)
Comment (MartinquaXD) [reply crates/autopilot/src/run.rs:232]: Yeah, the bad token detection still uses it. :/
I don't have a very concrete plan how to move forward with this. My idea was to use our currently strictest quote verification mode `enforce-if-possible` to phase out bad token detection. There are a bunch of caveats around this so I would just try it out on staging and see how it goes. 🙏

**Pull request merged**